File: nattybear/level1/하샤드-수/Solution.java
import java.util.ArrayList;

class Solution {
  public boolean solution(int x) {
    int sum = toDigits(x).stream().reduce(0, (acc, n) -> acc + n);
    return x % sum == 0;
  }

  static ArrayList<Integer> toDigits(int x) {
    ArrayList<Integer> digits = new ArrayList<>();
    while (x > 0) {
      digits.add(x % 10);
      x = x / 10;
    }
    return digits;
  }
}

File: nattybear/level1/하샤드-수/solution.js
function solution(x) {
  const sum = digits(x).reduce((acc, x) => acc + x)
  return x % sum == 0
}

function digits(x) {
  const _digits = []
  while (x > 0) {
    _digits.push(x % 10)
    x = Math.floor(x / 10)
  }
  return _digits
}
